src/flowchem/devices/knauer/knauer_autosampler_component.py

```diff
@@ -0,0 +1,314 @@
+from __future__ import annotations
+from flowchem import ureg
+from typing import TYPE_CHECKING
+from enum import Enum
+from typing import TYPE_CHECKING
+from loguru import logger
+
+
+if TYPE_CHECKING:
+    from .knauer_autosampler import KnauerAutosampler
+
+from flowchem.components.cnc.cnc import CNC
+from flowchem.components.pumps.syringe_pump import SyringePump
+from flowchem.components.valves.distribution_valves import FourPortDistributionValve
+from flowchem.components.valves.injection_valves import SixPortTwoPositionValve
+
+from flowchem.utils.exceptions import DeviceError
+try:
+    # noinspection PyUnresolvedReferences
+    from NDA_knauer_AS.knauer_AS import *
+
+    HAS_AS_COMMANDS = True
+except ImportError:
+    HAS_AS_COMMANDS = False
+
+
+class AutosamplerCNC(CNC):
+    """
+    Control a Knauer CNC component .
+
+    Attributes:
+        hw_device (KnauerAutosampler): The hardware device for the Knauer CNC component.
+    """
+
+    hw_device: KnauerAutosampler  # for typing's sake
+
+    def __init__(self, name: str, hw_device: KnauerAutosampler) -> None:
+        """Initialize component."""
+        super().__init__(name, hw_device)
+        self.add_api_route("/set_needle_position", self.set_needle_position, methods=["PUT"])
+
+
+    async def set_needle_position(self, position: str = "") -> None:
```

Comment:
this should be set_x_position. positions can then be columns on the plate or the special positions like wash waste etc

Reply:
reason is, that in cnc idea, we do not necessarily have a needle